lib/chef/deprecation/warnings.rb

```diff
@@ -24,8 +24,8 @@ def add_deprecation_warnings_for(method_names)
         method_names.each do |name|
           m = instance_method(name)
           define_method(name) do |*args|
-            Chef::Log.warn "Method '#{name}' of '#{self.class}' is deprecated. It will be removed in Chef 12."
-            Chef::Log.warn "Please update your cookbooks accordingly. Accessed from:"
+            Chef::Log.deprecation "Method '#{name}' of '#{self.class}' is deprecated. It will be removed in Chef 12."
+            Chef::Log.deprecation "Please update your cookbooks accordingly. Accessed from:"
```

Comment:
Because this can now raise an exception, these two strings should be combined into 1 call to `Chef::Log.deprecation`